### PR TITLE
Pavel/quick filters layout

### PR DESCRIPTION
**Pull request by galloween**
Powerfull new Quick Filter Layout component, that can be used in two ways:

- Consumer can provide (via ng-content) any number of form elements and manage ther bindings, events & functionality manually

- Consumer can provide the form elements, and [quickFilters] QuickFilterConfig array. Elements will be initialized according to QuickFilterConfig's. Change events from all components are combined into single (filtersChange) output. Form elements and/or QuickFilterConfig's can be added/updated dynamically.

This PR also includes:
- Some form elements updates, in particular - Social input, Datepicker, and some others
- Related Interfaces && CSS updates

### CODE REVIEW

File: projects/ui-framework/src/lib/buttons/text-button/text-button.component.ts [hunk 1]
```diff
@@ -16,16 +17,15 @@ import { LinkColor } from '../../buttons-indicators/link/link.enum';
...
 })
-export class TextButtonComponent {
+export class TextButtonComponent extends BaseButtonElement {
   @HostBinding('class.color-primary') get colorPrimary(): boolean {
     return this.color === LinkColor.primary;
   }
   @HostBinding('class.disabled') @Input() disabled = false;
 
-  @Input() text: string;
-  @Input() icon: Icons;
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/buttons/text-button/text-button.component.ts]: does the base form element have these inputs?

Comment (galloween) [reply projects/ui-framework/src/lib/buttons/text-button/text-button.component.ts]: BaseButtonElement has it.
text button was not BaseButtonElement before, not it is

File: projects/ui-framework/src/lib/form-elements/base-form-element.ts [hunk 2]
```diff
@@ -102,7 +102,11 @@ export abstract class BaseFormElement
         value
       );
     }
-    if (isNullOrUndefined(this.value) && this.baseValue !== undefined) {
+
+    if (
+      (value === undefined || isNullOrUndefined(this.value)) &&
+      this.baseValue !== undefined
+    ) {
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/base-form-element.ts]: add test

File: projects/ui-framework/src/lib/form-elements/date-range-picker/date-range-picker.component.spec.ts [hunk 3]
```diff
@@ -54,19 +54,19 @@ describe('DateRangePickerComponent', () => {
...
       declarations: [DateRangePickerComponent],
       providers: [
         { provide: UtilsService, useValue: utilsServiceStub },
         { provide: MobileService, useValue: mobileServiceStub },
         DateParseService,
-        EventManagerPlugins[0]
+        EventManagerPlugins[0],
       ],
-      schemas: [NO_ERRORS_SCHEMA]
+      schemas: [NO_ERRORS_SCHEMA],
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/date-range-picker/date-range-picker.component.spec.ts]: this should be removed, different pr

File: projects/ui-framework/src/lib/form-elements/datepicker/datepicker.abstract.ts [hunk 4]
```diff
@@ -91,9 +92,15 @@ export abstract class BaseDatepickerElement extends BaseFormElement
...
   readonly types = DatepickerType;
 
+  private doneFirstChange = false;
+
   protected doOnPickerOpen(picker: MatDatepicker<any>): void {}
 
   ngOnInit(): void {
+    if (!this.doneFirstChange) {
+      this.ngOnChanges({});
+    }
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/datepicker/datepicker.abstract.ts]: add test

File: projects/ui-framework/src/lib/form-elements/social/social.component.ts [hunk 5]
```diff
@@ -100,29 +104,29 @@ export class SocialComponent extends BaseFormElement
     }
   }
 
+  ngOnInit(): void {
+    this.inputId = this.bInput.id;
+    this.narrowInput =
+      (this.host.nativeElement as HTMLElement).offsetWidth < 300;
+  }
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/form-elements/social/social.component.ts]: add test

File: projects/ui-framework/src/lib/search/quick-filter-layout/quick-filter-layout.component.ts [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+        ));
+  }
+
+  private getChangeEmitter(formComp: BaseFormElement): EventEmitter<any> {
+    return formComp[
+      this.formCompEmittersMap[formComp.id] ||
+        this.findChangeEmitterKey(formComp)
+    ];
+  }
+}
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/search/quick-filter-layout/quick-filter-layout.component.ts]: add test for file

**Pull request merged**